src/vivarium/framework/results/manager.py

```diff
@@ -271,14 +266,22 @@ def register_binned_stratification(
             Keyword arguments for :meth: pandas.cut.
         """
 
-        def _bin_data(data: pd.Series | pd.DataFrame) -> pd.Series:
+        def _bin_data(data: pd.Series[int | float] | pd.DataFrame) -> pd.Series[Any]:
             """Use pandas.cut to bin continuous values"""
-            data = data.squeeze()
+            # mypy wants this squeeze to return a Series or DataFrame
+            if isinstance(data, pd.DataFrame):
+                data = data.squeeze()
```

Comment:
You should do 
```python
data = data.squeeze(axis=1)
```
This guarantees you won't get a scalar.

Reply:
mypy unfortunately still complains with this approach (with the original error and a new one about passing 1 to the axis argument) so I went with a one liner instead